### PR TITLE
Fix throughput example

### PR DESCRIPTION
**Pull request by shaladdle**
The reactor wasn't being run. Also, I get speeds of ~1MB/s on both my macbook and my linux machine, so I reduced the transfer size for the tarpc case to 24MB so it finishes in about 24s, instead of 4 minutes.

### CODE REVIEW

**Comment by tikue** [comment]
I'd rather just fix bincode. It's a 10 line patch, so hopefully they'd accept it.

File: examples/throughput.rs [hunk 1]
```diff
@@ -103,5 +111,5 @@ fn main() {
     let _ = env_logger::init();
     let _ = *BUF; // To non-lazily initialize it.
     bench_tcp(256 << 20);
-    bench_tarpc(256 << 20);
+    bench_tarpc(24 << 20);
```
Comment (tikue) [inline examples/throughput.rs]: Keep this 256. I'd rather just wait for bincode to fix their performance issues.

File: examples/throughput.rs [hunk 2]
```diff
@@ -55,12 +56,19 @@ impl FutureService for Server {
...
-    let mut client = SyncClient::connect(addr, client::Options::default()).unwrap();
+    let (tx, rx) = mpsc::channel();
+    thread::spawn(move || {
+        let mut reactor = reactor::Core::new().unwrap();
+        let addr = Server.listen("localhost:0".first_socket_addr(),
+                    &reactor.handle(),
+                    server::Options::default())
+            .unwrap();
+        tx.send(addr).unwrap();
+        loop {
```
Comment (tikue) [inline examples/throughput.rs]: change to `reactor.run(server);`

File: examples/throughput.rs [hunk 3]
```diff
@@ -55,12 +56,19 @@ impl FutureService for Server {
...
-    let reactor = reactor::Core::new().unwrap();
-    let addr = Server.listen("localhost:0".first_socket_addr(),
-                &reactor.handle(),
-                server::Options::default())
-        .unwrap();
-    let mut client = SyncClient::connect(addr, client::Options::default()).unwrap();
+    let (tx, rx) = mpsc::channel();
+    thread::spawn(move || {
+        let mut reactor = reactor::Core::new().unwrap();
+        let addr = Server.listen("localhost:0".first_socket_addr(),
```
Comment (tikue) [inline examples/throughput.rs]: change to `let (addr, server) = ...`

Comment (shaladdle) [inline examples/throughput.rs]: Done

File: examples/throughput.rs (hunk 2 above)
Comment (shaladdle) [inline examples/throughput.rs]: Done

File: examples/throughput.rs (hunk 1 above)
Comment (shaladdle) [inline examples/throughput.rs]: Done

**Comment by TyOverby** [comment]
You'll likely get even better perf once https://github.com/TyOverby/bincode/pull/103 lands and you can avoid doing a byte-shuffle on each read.

**Pull request merged**